tools/cmake/upload_methods/install_bin_file.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+# Bin file installer that uses mbed tools to find the board, install the file, and reset.
+# Code extracted from mbed-cli.
+
+# usage:
+
+import os
+import ctypes
+import sys
+
+from mbed_os_tools.test.host_tests_toolbox import flash_dev, reset_dev
+import mbed_os_tools.detect
+
+
+def get_detected_targets():
+    targets = []
+    oldError = None
+    if os.name == 'nt':
+        oldError = ctypes.windll.kernel32.SetErrorMode(1) # Disable Windows error box temporarily. note that SEM_FAILCRITICALERRORS = 1
+    mbeds = mbed_os_tools.detect.create()
+    detect_muts_list = mbeds.list_mbeds()
+    if os.name == 'nt':
+        ctypes.windll.kernel32.SetErrorMode(oldError)
+
+    for mut in detect_muts_list:
+        targets.append({
+            'id': mut['target_id'], 'name': mut['platform_name'],
+            'mount': mut['mount_point'], 'serial': mut['serial_port'],
+            'uid': mut['target_id_usb_id']
+        })
+
+    return targets
+
+def error(lines, code=-1):
+    sys.stderr.write("[install-bin-file] ERROR: %s\n" % (lines.pop(0),))
+    for line in lines:
+        sys.stderr.write("       %s\n" % (line,))
+    sys.stderr.write("---\n")
+    sys.exit(code)
+
+if len(sys.argv) < 3:
+    print("Error: Usage: python " + sys.argv[0] + " <path to bin file> <Mbed OS target name> <serial baudrate for sending break> [Target UID for distinguishing multiple targets]")
+    exit(1)
```

Comment:
Switched to sys.exit.  I don't think this needs really sophisticated argument handling though because it's only ever called by CMake scripts, not manually.